### PR TITLE
Vertx 3.8.3 step4

### PR DESCRIPTION
**Pull request by sophokles73**
Fix usage of deprecated API in MQTT, Kura and HTTP adapters and Device Connection service

### CODE REVIEW

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java [hunk 1]
```diff
@@ -462,23 +467,23 @@ public final void doStop(final Future<Void> stopFuture) {
...
             insecureServer.close(insecureServerStopTracker);
         } else {
             insecureServerStopTracker.complete();
         }
 
-        CompositeFuture.all(serverStopTracker, insecureServerStopTracker)
-            .compose(v -> postShutdown())
-            .compose(s -> stopFuture.complete(), stopFuture);
+        CompositeFuture.all(serverStopTracker.future(), insecureServerStopTracker.future())
+        .compose(v -> postShutdown())
```
Comment (ctron) [inline adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: Something with the indentation here looks wrong.

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java [hunk 2]
```diff
@@ -975,29 +1014,45 @@ protected void setNonEmptyResponsePayload(final HttpServerResponse response, fin
...
-                                            command.getPayloadSize(),
-                                            commandSample);
-                                }
-                                cancelCommandReceptionTimer(ctx);
-                                setTtdStatus(ctx, TtdStatus.COMMAND);
-                                responseReady.tryComplete();
-                            });
                 }
+
+//                if (requestProcessed.get()) {
```
Comment (ctron) [inline adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: Do we need this huge block of commented out code?

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java (hunk 1 above)
Comment (sophokles73) [reply adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: hm, this is how Eclipse indents (or rather doesn't indent) it when I hit return at the end of the former line ...

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java (hunk 2 above)
Comment (sophokles73) [reply adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: definitely not :+1:

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java (hunk 1 above)
Comment (ctron) [reply adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: That is weird. Try formatting the code, every now and then I see differences between direct editing and formatting afterwards.

**Pull request merged**